### ISSUE TITLE
Neighborhood search not working for Ellipsoid

### ISSUE THREAD

**Issue by hameye** [JuliaEarth/GeoStats.jl#176]
```julia
julia> model =
           SphericalVariogram(range = 30.0, sill = 1.0) +
           SphericalVariogram(range = 100.0, sill = 2.0) +
           NuggetEffect(3.0)
NestedVariogram{3}
  structures
    └─SphericalVariogram(range=30.0, sill=1.0, nugget=0.0)
    └─SphericalVariogram(range=100.0, sill=2.0, nugget=0.0)
    └─NuggetEffect(range=0.0, sill=3.0, nugget=3.0)
  coefficients
    └─1
    └─1
    └─1

julia> solver = Kriging(:Z => (variogram = model,maxneighbors=20,neighborhood=Ellipsoid([20., 20],[pi/2])))
Kriging
  └─Z
    └─variogram ⇨ NestedVariogram{3}
    └─minneighbors ⇨ 1
    └─maxneighbors ⇨ 20
    └─neighborhood ⇨ Ellipsoid{Float64}([20.0, 20.0], [1.5707963267948966], TaitBryanExtr)
    └─distance ⇨ Euclidean(0.0)


julia> solution = solve(problem, solver)
ERROR: MethodError: no method matching radius(::Ellipsoid{Vector{Float64}, Vector{Float64}, DataType, Distances.Mahalanobis{StaticArrays.SMatrix{2, 2, Float64, 4}}})
Closest candidates are:
  radius(::Ball) at /home/hadrienmeyer/.julia/dev/Meshes/src/primitives/ball.jl:20
  radius(::Sphere) at /home/hadrienmeyer/.julia/dev/Meshes/src/primitives/sphere.jl:39
  radius(::Cylinder) at /home/hadrienmeyer/.julia/dev/Meshes/src/primitives/cylinder.jl:22
  ...
Stacktrace:
 [1] search!(neighbors::Vector{Int64}, pₒ::Point2, method::KBallSearch{GeoData{PointSet{2, Float64}, TypedTables.Table{NamedTuple{(:Z,), Tuple{Float64}}, 1, NamedTuple{(:Z,), Tuple{Vector{Float64}}}}}, Ellipsoid{Vector{Float64}, Vector{Float64}, DataType, Distances.Mahalanobis{StaticArrays.SMatrix{2, 2, Float64, 4}}}, NearestNeighbors.BallTree{StaticArrays.SVector{2, Float64}, 2, Float64, Distances.Mahalanobis{StaticArrays.SMatrix{2, 2, Float64, 4}}}}; mask::Nothing)
   @ Meshes ~/.julia/dev/Meshes/src/neighborsearch/kball.jl:32
 [2] search!(neighbors::Vector{Int64}, pₒ::Point2, method::KBallSearch{GeoData{PointSet{2, Float64}, TypedTables.Table{NamedTuple{(:Z,), Tuple{Float64}}, 1, NamedTuple{(:Z,), Tuple{Vector{Float64}}}}}, Ellipsoid{Vector{Float64}, Vector{Float64}, DataType, Distances.Mahalanobis{StaticArrays.SMatrix{2, 2, Float64, 4}}}, NearestNeighbors.BallTree{StaticArrays.SVector{2, Float64}, 2, Float64, Distances.Mahalanobis{StaticArrays.SMatrix{2, 2, Float64, 4}}}})
   @ Meshes ~/.julia/dev/Meshes/src/neighborsearch/kball.jl:31
 [3] solve_approx(problem::EstimationProblem{GeoData{PointSet{2, Float64}, TypedTables.Table{NamedTuple{(:Z,), Tuple{Float64}}, 1, NamedTuple{(:Z,), Tuple{Vector{Float64}}}}}, CartesianGrid{2, Float64}, 1}, var::Symbol, preproc::Dict{Symbol, NamedTuple})
   @ GeoEstimation ~/.julia/dev/GeoEstimation/src/kriging.jl:181
 [4] solve(problem::EstimationProblem{GeoData{PointSet{2, Float64}, TypedTables.Table{NamedTuple{(:Z,), Tuple{Float64}}, 1, NamedTuple{(:Z,), Tuple{Vector{Float64}}}}}, CartesianGrid{2, Float64}, 1}, solver::Kriging)
   @ GeoEstimation ~/.julia/dev/GeoEstimation/src/kriging.jl:143
 [5] top-level scope
   @ REPL[31]:1

```

**Comment by juliohm** [comment]
As we discussed over Gitter, the issue can be fixed in the `KBallSearch` by repeating the same logic we implemented in the `NeighborhoodSearch` [here](https://github.com/JuliaGeometry/Meshes.jl/blob/a3598f0d0a4c65e2179f8fd5da0e551a0f4d698f/src/neighborsearch/neighborhood.jl#L30-L34). As you pointed out, the issue occurs because an `Ellipsoid` doesn't have a radius and `KBallSearch` is trying to call for the radius of the neighborhood.

An easy quick fix consists of copying the code linked above from `NeighborhoodSearch` to `KBallSearch` with minor changes. Appreciate your help to fix that. :)
